src/control-plane/backend/lambda/api/router/environment.ts

```diff
@@ -13,20 +13,23 @@
 
 import express from 'express';
 import { body, header, query } from 'express-validator';
-import { defaultAssumeRoleTypeValid, defaultPageValueValid, defaultRegionValueValid, defaultSubnetTypeValid, isRequestIdExisted, isValidEmpty, validate } from '../common/request-valid';
+import { defaultAssumeRoleTypeValid, defaultPageValueValid, defaultRegionValueValid, defaultSubnetTypeValid, isRequestIdExisted, isValidEmpty, validate, validateRole } from '../common/request-valid';
+import { IUserRole } from '../common/types';
 import { EnvironmentServ } from '../service/environment';
 
 const router_env = express.Router();
 const environmentServ: EnvironmentServ = new EnvironmentServ();
 
 router_env.get(
   '/regions',
+  validateRole([IUserRole.ADMIN, IUserRole.OPERATOR]),
```

Comment:
Is there a way we do validation in a middleware to check the request path with role without validating in every route?

Reply:
if use middleware, also need a `map<api route, roles>` to manage this relationship.

Reply:
I think we should also organize the API route with patterns,

- `op/*` for pipeline operation
- `analyze/*` for analyst's explorative analysis
- `users/*` for the administrator to manage the users and roles

Then, we can benefit from it by granting the roles with corresponding permissions.

Reply:
done